example/test.py

```diff
@@ -353,6 +372,77 @@ def test_gzip_small_file_not_compressed(self, web_container, web_client):
         ))
 
 
+class TestNginx(object):
+    def test_expected_processes(self, nginx_only_container):
+        """
+        When the container is running, there should be 2 running processes:
+        the Nginx master and worker processes.
+        """
+        ps_rows = nginx_only_container.list_processes()
+        ps_tree = build_process_tree(ps_rows)
+
+        nginx_master_args = 'nginx: master process nginx -g daemon off;'
+        nginx_worker_args = 'nginx: worker process'
+
+        assert_that(
+            ps_tree,
+            MatchesPsTree('root', nginx_master_args, pid=1, children=[
+                MatchesPsTree('nginx', nginx_worker_args),
+            ]))
+
+    def test_nginx_access_logs(self, nginx_container, web_container):
```

Comment:
Whether a test is "for" the Nginx container or for the web/gunicorn container is kinda hard to say, since they work together. It's possible that it makes more sense for all the tests for headers and such to be in `TestNginx` instead of `TestWeb`, but I'm not sure.